contracts/crowdfund/Crowdfund.sol

```diff
@@ -585,6 +613,29 @@ abstract contract Crowdfund is Implementation, ERC721Receiver, CrowdfundNFT {
         }
     }
 
+    //Allows contributor to rage quit before crowdfund is finalized.
+    function _rageQuit(address payable contributor) private {
```

Comment:
I was trying to follow how `_contribute()` is called with `msg.sender` being passed, I'm not clear on why that is working because shouldn't `msg.sender == proxyContract` in `_contribute()`? When I try to set `contributor == msg.sender` in `rageQuit()` it is not the correct address

Reply:
Not quite sure what you mean but `msg.sender` shouldn't be the proxy contract, it should be the user calling the function.

> When I try to set contributor == msg.sender in rageQuit() it is not the correct address

Could you show me how you are doing this?

Reply:
In the Foundry test, unless you are using `vm.prank()`, it should be the test contract address itself that is `msg.sender` when you call `rageQuit()`. If you are using `vm.prank()`, it is whatever address you are "pranking" as that should `msg.sender`. Is this not the case when you run tests?

Reply:
I am pranking as the contributor, here's the current code...
![Screenshot 2023-02-21 at 1 23 44 PM](https://user-images.githubusercontent.com/5912724/220450646-228c5552-8ae8-4681-9323-558b45ea164c.png)

![Screenshot 2023-02-21 at 1 24 29 PM](https://user-images.githubusercontent.com/5912724/220450725-3090d29d-b760-4527-9f70-d625f016eb1a.png)

Reply:
By the time it gets to `rageQuit()`, the test is not pranking as the contributor anymore because `vm.prank()` only works for one call.

Reply:
Oh, thank you, I had incorrectly thought it called as the user pranking until `vm.stop()` was called